src/llmcompressor/transformers/compression/sparsity_config.py

```diff
@@ -95,11 +98,12 @@ def from_pretrained(
         sparsity_structure = SparsityConfigMetadata.infer_sparsity_structure(
             model=model
         )
-        if is_model_quantized(model):
-            # compressing a sparse quantized model is not supported yet
+        if quantization_format == CompressionFormat.marlin_24:
+            # sparse compressor should be dense for marlin
+            # compression
             format = CompressionFormat.dense.value
-        elif compress:
-            format = CompressionFormat.sparse_bitmask.value
+        elif compress and sparsity_structure == SparsityStructure.TWO_FOUR.value:
+            format = CompressionFormat.sparse_24_bitmask.value
```

Comment:
Add a note, that this is only supported for int8, fp8 cases with 2:4 sparsity + cutlass supports weight + activation quantization only, add assertions for this;